bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/ClientLibraryImplTest.java

```diff
@@ -15,94 +15,126 @@
+    private static final String CLIENT_LIB_JS = TEST_ROOT_PAGE + "/clientlib-js";
+    private static final String TEST_APPS_ROOT = "/apps/core-components-examples/clientlibs";
+
+    protected final AemContext context = CoreComponentTestContext.newAemContext();
+
 
     @BeforeEach
     void setUp() {
-        this.testLogger = TestLoggerFactory.getTestLogger(ClientLibraryImpl.class);
+        context.load().json(TEST_BASE + CoreComponentTestContext.TEST_CONTENT_JSON, TEST_ROOT_PAGE);
+        context.load().json(TEST_BASE + CoreComponentTestContext.TEST_APPS_JSON, TEST_APPS_ROOT);
 
-        initMocks(this);
     }
 
-    @AfterEach
-    void tearDown() {
-        TestLoggerFactory.clear();
+    @Test
+    void testEmptyGetInlineCss() throws Exception {
+        context.registerInjectActivateService(new MockHtmlLibraryManager(mock(com.adobe.granite.ui.clientlibs.ClientLibrary.class)));
+        context.registerInjectActivateService(new ClientLibraryAggregatorServiceImpl());
+
+        ClientLibrary clientLibrary = getClientLibraryUnderTestFromResource(CLIENT_LIB_CSS);
+
+        assertEquals("", clientLibrary.getInline());
     }
 
     @Test
-    public void getInline() {
-        this.cl.categories = this.categories;
-        this.cl.type = this.type;
+    void testEmptyGetInlineJs() throws Exception {
+        context.registerInjectActivateService(new MockHtmlLibraryManager(mock(com.adobe.granite.ui.clientlibs.ClientLibrary.class)));
+        context.registerInjectActivateService(new ClientLibraryAggregatorServiceImpl());
+        ClientLibrary clientLibrary = getClientLibraryUnderTestFromResource(CLIENT_LIB_JS);
 
-        when(this.clientLibraryAggregatorServiceMock.getClientLibOutput(this.categories, this.type))
-          .thenReturn("output inline");
+        assertEquals("", clientLibrary.getInline());
+    }
+
+    @Test
+    void testGetInlineCss() throws Exception {
+        HtmlLibraryManager htmlLibraryManager = context.registerInjectActivateService(mock(MockHtmlLibraryManager.class));
+        context.registerInjectActivateService(new ClientLibraryAggregatorServiceImpl());
+        HtmlLibrary library = mock(HtmlLibrary.class);
 
-        assertEquals("output inline", this.cl.getInline());
+        ClientLibrary clientLibrary = getClientLibraryUnderTestFromResource(CLIENT_LIB_CSS);
+        Resource clientLibResource = context.currentResource(TEST_APPS_ROOT + "/clientlib-base-amp/styles/index.css");
+
+        when(htmlLibraryManager.getLibraries(any(), any(), anyBoolean(), anyBoolean())).thenReturn(Arrays.asList(mock(com.adobe.granite.ui.clientlibs.ClientLibrary.class)));
+        when(htmlLibraryManager.getLibrary(any(), any())).thenReturn(library);
+        when(library.getInputStream(anyBoolean())).thenReturn(clientLibResource.adaptTo(InputStream.class));
+
+        String outputString = IOUtils.toString(clientLibResource.adaptTo(InputStream.class), StandardCharsets.UTF_8);
```

Comment:
this is doing the same scheme of 'when i do this return that' and verifying that against itself. 
If possible you should have the actual expected output in a hardcoded string so that you aren't relying on your resource adaptTo to return the same thing twice.

Reply:
That makes sense and updated the outputString to be a hardcoded string value.